ADAL/src/cache/ADTokenCacheItem+Internal.m

```diff
@@ -196,29 +219,13 @@ - (BOOL)fillItemWithResponse:(NSDictionary*)responseDictionary
     
     [self fillExpiration:responseDictionary];
     
-    BOOL isMRRT = ![NSString adIsStringNilOrBlank:[responseDictionary objectForKey:OAUTH2_RESOURCE]] && ![NSString adIsStringNilOrBlank:self.refreshToken];
+    BOOL isMRRT = ![NSString adIsStringNilOrBlank:self.resource] && ![NSString adIsStringNilOrBlank:self.refreshToken];
```

Comment:
If you have to move up fill expiration then do it, but this change has the potential to cause unexpected side effects.